xwiki-platform-core/xwiki-platform-query/xwiki-platform-query-xwql/src/test/java/org/xwiki/query/xwql/internal/XWQLtoHQLTranslatorTest.java

```diff
@@ -249,15 +249,28 @@ public void testLists() throws Exception
         // DBStringListProperty
         assertTranslate("from doc.object('XWiki.ArticleClass') as a where :cat member of a.category",
             "select doc.fullName from XWikiDocument as doc , BaseObject as a , DBStringListProperty as a_category1" +
-                " where ( :cat in elements( a_category1.list ) ) and doc.fullName=a.name and a.className='XWiki.ArticleClass' and a_category1.id.id=a.id and a_category1.id.name='category'");
+                " join a_category1.list a_category1list where ( :cat in elements( a_category1.list ) )" +
+                " and doc.fullName=a.name and a.className='XWiki.ArticleClass' and a_category1.id.id=a.id" +
+                " and a_category1.id.name='category'");
```

Comment:
stupid question: in term of performance, does the new way have any impact?

Reply:
With this XWQL query:
```sql
select obj.staticList1 from Document as doc, doc.object(Help.Applications.Movies.Code.MoviesClass) as obj
```
The previously generated HQL query was:
```sql
select elements(obj_staticList11.list) from com.xpn.xwiki.doc.XWikiDocument as doc, com.xpn.xwiki.objects.BaseObject as obj, com.xpn.xwiki.objects.DBStringListProperty as obj_staticList11 where doc.fullName=obj.name and obj.className='Help.Applications.Movies.Code.MoviesClass' and obj_staticList11.id.id=obj.id and obj_staticList11.id.name='staticList1'
```
which then generates to this SQL query:
```sql
select list3_.XWL_VALUE as col_0_0_ from xwikidoc xwikidocum0_ cross join xwikiobjects baseobject1_ cross join xwikilists dbstringli2_ inner join xwikiproperties dbstringli2_1_ on dbstringli2_.XWL_ID=dbstringli2_1_.XWP_ID and dbstringli2_.XWL_NAME=dbstringli2_1_.XWP_NAME, xwikilistitems list3_ cross join xwikilistitems list4_ where dbstringli2_.XWL_ID=list3_.XWL_ID and dbstringli2_.XWL_NAME=list3_.XWL_NAME and dbstringli2_.XWL_ID=list4_.XWL_ID and dbstringli2_.XWL_NAME=list4_.XWL_NAME and 1=1 and xwikidocum0_.XWD_FULLNAME=baseobject1_.XWO_NAME and baseobject1_.XWO_CLASSNAME='Help.Applications.Movies.Code.MoviesClass' and dbstringli2_.XWL_ID=baseobject1_.XWO_ID and dbstringli2_.XWL_NAME='staticList1' 
```
The new generated HQL query is now:
```sql
 select obj_staticList11list from com.xpn.xwiki.doc.XWikiDocument as doc , com.xpn.xwiki.objects.BaseObject as obj , com.xpn.xwiki.objects.DBStringListProperty as obj_staticList11 join obj_staticList11.list obj_staticList11list WHERE 1=1  and doc.fullName=obj.name and obj.className='Help.Applications.Movies.Code.MoviesClass' and obj_staticList11.id.id=obj.id and obj_staticList11.id.name='staticList1'
```
which leads to this SQL query:
```sql
select list3_.XWL_VALUE as col_0_0_ from xwikidoc xwikidocum0_ cross join xwikiobjects baseobject1_ cross join xwikilists dbstringli2_ inner join xwikiproperties dbstringli2_1_ on dbstringli2_.XWL_ID=dbstringli2_1_.XWP_ID and dbstringli2_.XWL_NAME=dbstringli2_1_.XWP_NAME inner join xwikilistitems list3_ on dbstringli2_.XWL_ID=list3_.XWL_ID and dbstringli2_.XWL_NAME=list3_.XWL_NAME where 1=1 and xwikidocum0_.XWD_FULLNAME=baseobject1_.XWO_NAME and baseobject1_.XWO_CLASSNAME='Help.Applications.Movies.Code.MoviesClass' and dbstringli2_.XWL_ID=baseobject1_.XWO_ID and dbstringli2_.XWL_NAME='staticList1'
```

Reply:
ok but the question is whether it has any perf implication or not.

Reply:
How can I measure that? I don't think there will be great perf implication though.

Reply:
The simplest to me is to take MySQL and do an EXPLAIN with the 2 queries and/or simply verify exec times.